actors/miner/tests/extend_sector_expiration_test.rs

```diff
@@ -352,3 +365,413 @@ fn supports_extensions_off_deadline_boundary() {
+
+    // assert that new expiration exists
+    let (_, mut partition) = h.get_deadline_and_partition(&rt, deadline_index, partition_index);
+    let expiration_set = partition.pop_expired_sectors(rt.store(), expiration - 1, quant).unwrap();
+    assert!(expiration_set.is_empty());
+
+    let expiration_set =
+        partition.pop_expired_sectors(rt.store(), quant.quantize_up(expiration), quant).unwrap();
+    assert_eq!(expiration_set.len(), 1);
+    assert!(expiration_set.on_time_sectors.get(sector_number));
+
+    h.check_state(&rt);
+}
+
+fn make_fake_claim(
+    claim_id: u64,
+    sector: &SectorOnChainInfo,
+    client: ActorID,
+    provider: ActorID,
+    new_expiration: ChainEpoch,
+    deal: &TestVerifiedDeal,
+    term_min: ChainEpoch,
+) -> FILPlusClaim {
+    FILPlusClaim {
+        provider,
+        client,
+        data: make_piece_cid(format!("piece for claim {}", claim_id).as_bytes()),
+        size: deal.space,
+        term_min,
+        term_max: new_expiration - sector.activation,
+        term_start: sector.activation,
+        sector: sector.sector_number,
+    }
+}
+
+#[test]
+fn update_expiration_with_claim() {
+    let (mut h, mut rt) = setup();
+    // add in verified deal
+    let verified_deals = vec![TestVerifiedDeal { space: PaddedPieceSize(h.sector_size as u64) }];
+    let old_sector = commit_sector_verified_deals(&verified_deals, &mut h, &mut rt);
+    h.advance_and_submit_posts(&mut rt, &vec![old_sector.clone()]);
+
+    let state: State = rt.get_state();
+
+    let (deadline_index, partition_index) =
+        state.find_sector(rt.policy(), rt.store(), old_sector.sector_number).unwrap();
+
+    let extension = 42 * rt.policy().wpost_proving_period;
+    let new_expiration = old_sector.expiration + extension;
+
+    let claim_id = 400;
+    let client = Address::new_id(3000).id().unwrap();
+    let fake_claim = make_fake_claim(
+        claim_id,
+        &old_sector,
+        client,
+        h.receiver.id().unwrap(),
+        new_expiration,
+        &verified_deals[0],
+        rt.policy.minimum_verified_allocation_term,
+    );
+    let mut claims = HashMap::new();
+    claims.insert(claim_id, Ok(fake_claim));
+
+    let params = ExtendSectorExpiration2Params {
+        extensions: vec![ExpirationExtension2 {
+            deadline: deadline_index,
+            partition: partition_index,
+            sectors: make_bitfield(&[]),
+            new_expiration,
+            sectors_with_claims: vec![SectorClaim {
+                sector_number: old_sector.sector_number,
+                maintain_claims: vec![claim_id],
+            }],
+        }],
+    };
+
+    h.extend_sectors2(&mut rt, params, claims).unwrap();
+
+    // assert sector expiration is set to the new value
+    check_for_expiration(
+        &mut h,
+        &mut rt,
+        new_expiration,
+        old_sector.sector_number,
+        deadline_index,
+        partition_index,
+    );
+}
```

Comment:
This test seems redundant given the subsequent one. What's it uniquely exercising?